src/ipc.cpp

```diff
@@ -21,12 +21,14 @@ std::vector<std::ostream*>  g_logging_err_outs = {
 };
 
 #define IPC_JSON_READ(ROOT) \
-	{ \
-		Json::Reader  reader; \
-		if (!reader.parse(std::string(buf->payload, buf->header->size), ROOT, false)) { \
-			throw invalid_reply_payload_error(auss_t() << "Failed to parse reply on \"" i3IPC_TYPE_STR "\": " << reader.getFormattedErrorMessages()); \
-		} \
-	}
+{ \
+	Json::CharReaderBuilder b; \
+	Json::CharReader* reader(b.newCharReader()); \
```

Comment:
Doesn't this leak memory?

Reply:
Yeah, probably does. I'm gonna open a PR